src/main/java/com/bookstore/repository/BookRepository.java

```diff
@@ -10,6 +10,9 @@
 
 @Repository
 public interface BookRepository extends JpaRepository<Book, Long> {
-    @Query("select b from Book b inner join fetch b.categories c where c.id = :categoryId")
-    List<Book> findAllByCategoryId(Pageable pageable, @Param("categoryId") Long categoryId);
+    @Query("select b from Book b left join fetch b.categories c where c.id = :categoryId")
+    Page<Book> findAllByCategoryId(Pageable pageable, @Param("categoryId") Long categoryId);
+
+    @Query("select b from Book b left join fetch b.categories")
```

Comment:
let's use entitygraph here instead. Also, if you decide to write custom queries, let's make them more readable, use UPPERCASED keywords etc